fiftyone/core/odm/sample.py

```diff
@@ -232,9 +232,9 @@ def __setattr__(self, name, value):
         super().__setattr__(name, value)
 
     @property
-    def dataset_name(self):
-        """The name of the dataset to which this sample belongs, or ``None`` if
-        it has not been added to a dataset.
+    def collection_name(self):
```

Comment:
@tylerganter what do you think about making this private? Users shouldn't need to be concerned with the internal-facing name of the sample collection

Reply:
Isn't this entire class effectively private? I think making it public helps convey that this is something publicly available for use by `fiftyone.core.sample.Sample`

Reply:
Good point. I see that you did what I actually should have requested, which was just that this field is not exposed publicly on `fiftyone.core.sample.Sample`